src/pkg/docs/service/service.go

```diff
@@ -110,6 +110,115 @@ func (d *DefaultDocumentService) GetDocIndexByDocID(userID, docID string, ehrUUI
 	return nil, errors.ErrIsNotExist
 }
 
+func (d *DefaultDocumentService) GetLastVersionDocIndexByBaseId(userId, ehrId, baseDocumentId string, documentType types.DocumentType) (documentMeta *model.DocumentMeta, err error) {
+
+	documentsMeta, err := d.getDocIndexesByDocId(userId, ehrId, baseDocumentId, documentType)
+	if err != nil {
+		return nil, err
+	}
+
+	var lastVersion *semver.Version
+	for _, currentDocumentMeta := range documentsMeta {
+		v, err := semver.NewVersion(currentDocumentMeta.Version)
+		if err != nil {
+			return nil, err
+		}
+
+		if documentMeta == nil || v.GreaterThan(lastVersion) {
+			documentMeta = currentDocumentMeta
+			lastVersion = v
+		}
+	}
+
+	return documentMeta, nil
+}
+
+func (d *DefaultDocumentService) GetDocIndexByBaseIdAndVersion(userID string, ehrUUID *uuid.UUID, baseDocumentId, version string, documentType types.DocumentType) (documentMeta *model.DocumentMeta, err error) {
+	documentsMeta, err := d.getDocIndexesByDocId(userID, ehrUUID.String(), baseDocumentId, documentType)
```

Comment:
Search documents by full versionUid without separation baseDocumentId and version